### ISSUE TITLE
dm online validation could be broken when binlog reader reconnect when not using gtid

### ISSUE THREAD

**Issue by hihihuhu** [pingcap/tiflow#8335]
### What did you do?

binlog reader for dm online validation has `DisableRetrySync` set to true because https://github.com/pingcap/tiflow/blob/master/dm/syncer/util.go#L213, this could lead to `Err: table id xxx: invalid table id, no corresponding table map event` error, if the connection is broken in the middle of a transaction.

when this happens, the online validation will stuck forever

### What did you expect to see?

the online validation could handle binlog broken and continue

### What did you see instead?

assume this is the binlog
```
SET TIMESTAMP=1676594544/*!*/;
BEGIN
/*!*/;
# at 38845389
#230217  0:42:24 server id 900099330  end_log_pos 38845467 CRC32 0xf6701849     Table_map: `xxx`.`xxx` mapped to number 140
# at 38845467
#230217  0:42:24 server id 900099330  end_log_pos 38845538 CRC32 0x0357e7c9     Delete_rows: table id 140 flags: STMT_END_F

...

#230217  0:42:24 server id 900099330  end_log_pos 38845569 CRC32 0xeb59e74c     Xid = 149603988583
COMMIT/*!*/;
```
then in dm-worker log
```
[2023/02/18 02:58:10.907 +00:00] [ERROR] [streamer_controller.go:367] ["meet error when get binlog event"] [task=tidbmigration-sample-task] [unit="continuous validator"] [error="Header &replication.EventHeader{Timestamp:0x63eecd70, EventType:0x20, ServerID:0x35a66d02, EventSize:0x47, LogPos:0x250bc62, Flags:0x0}, Data \"....\", Err: table id 140: invalid table id, no corresponding table map event"] [errorVerbose="Header &replication.EventHeader{Timestamp:0x63eecd70, EventType:0x20, ServerID:0x35a66d02, EventSize:0x47, LogPos:0x250bc62, Flags:0x0}, Data \"\....\", Err: table id 140: invalid table id, no corresponding table map event\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).parseEvent\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/replication/binlogsyncer.go:782\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).onStream\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/replication/binlogsyncer.go:748\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2023/02/18 02:58:10] [info] binlogsyncer.go:196 syncer is closing...
[2023/02/18 02:58:10.877 +00:00] [INFO] [streamer_controller.go:109] ["last slave connection"] [task=tidbmigration-sample-task] [unit="continuous validator"] ["connection ID"=54712275]
[2023/02/18 02:58:10] [info] binlogsyncer.go:893 kill last connection id 54711567
[2023/02/18 02:58:10] [info] binlogsyncer.go:226 syncer is closed
[2023/02/18 02:58:10] [info] binlogsyncer.go:400 begin to sync binlog from position (mysql-bin-changelog.058161, 38845467)
[2023/02/18 02:58:10] [error] binlogsyncer.go:887 kill connection 54711567 error ERROR 1094 (HY000): Unknown thread id: 54711567
[2023/02/18 02:58:10] [info] binlogsyncer.go:164 create BinlogSyncer with config {429546523 mysql xxxxx xxxxxx    false false <nil> false +00:00 false 0 30s 1m0s 1 true true 0 <nil> 0xc01a641f20 0x74e860}
[2023/02/18 02:58:10] [error] binlogstreamer.go:78 close sync with err: io.CopyN failed. err unexpected EOF, copied 0, expected 66: connection was bad
[2023/02/18 02:58:10] [error] binlogsyncer.go:697 io.CopyN failed. err unexpected EOF, copied 0, expected 66: connection was bad
[2023/02/18 02:58:10.866 +00:00] [ERROR] [streamer_controller.go:367] ["meet error when get binlog event"] [task=tidbmigration-sample-task] [unit="continuous validator"] [error="io.CopyN failed. err unexpected EOF, copied 0, expected 66: connection was bad"] [errorVerbose="connection was bad\ngithub.com/go-mysql-org/go-mysql/mysql.init\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/mysql/error.go:10\nruntime.doInit\n\truntime/proc.go:6222\nruntime.doInit\n\truntime/proc.go:6199\nruntime.doInit\n\truntime/proc.go:6199\nruntime.doInit\n\truntime/proc.go:6199\nruntime.doInit\n\truntime/proc.go:6199\nruntime.main\n\truntime/proc.go:233\nruntime.goexit\n\truntime/asm_amd64.s:1571\nio.CopyN failed. err unexpected EOF, copied 0, expected 66\ngithub.com/go-mysql-org/go-mysql/packet.(*Conn).ReadPacketTo\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/packet/conn.go:168\ngithub.com/go-mysql-org/go-mysql/packet.(*Conn).ReadPacketReuseMem\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/packet/conn.go:97\ngithub.com/go-mysql-org/go-mysql/packet.(*Conn).ReadPacket\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/packet/conn.go:89\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).onStream\n\tgithub.com/go-mysql-org/go-mysql@v1.6.1-0.20220718092400-c855c26b37bd/replication/binlogsyncer.go:688\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2023/02/18 02:58:10.866 +00:00] [INFO] [streamer_controller.go:234] ["reset replication syncer"] [task=tidbmigration-sample-task] [unit="continuous validator"] [location="position: (mysql-bin-changelog.058161, 38845467), gtid-set: "]
[2023/02/18 02:58:10] [warn] binlogsyncer.go:707 retry sync is disabled
[2023/02/18 02:58:10] [info] binlogsyncer.go:196 syncer is closing...
[2023/02/18 02:58:10.272 +00:00] [INFO] [data_validator.go:1008] ["persist checkpoint and intermediate data"] [task=tidbmigration-sample-task] [unit="continuous validator"] ["pending size"=0] ["pending count"=0] ["new error"=0]
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
v6.2.0
```
Upstream MySQL/MariaDB server version:
```console
aurora-mysql-5.7
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
v6.2.0
```
How did you deploy DM: tiup or manually?
```console
tidb-operator
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "sourceStatus": {
                "source": "tidbmigration-sample-src",
                "worker": "test-single-cell-dm-worker-0",
                "result": null,
                "relayStatus": null
            },
            "subTaskStatus": [
                {
                    "name": "tidbmigration-sample-task",
                    "stage": "Running",
                    "unit": "Sync",
                    "result": null,
                    "unresolvedDDLLockID": "",
                    "sync": {
                        "totalEvents": "90872497",
                        "totalTps": "229",
                        "recentTps": "198",
                        "masterBinlog": "(mysql-bin-changelog.058406, 117618630)",
                        "masterBinlogGtid": "",
                        "syncerBinlog": "(mysql-bin-changelog.058406, 117613841)",
                        "syncerBinlogGtid": "",
                        "blockingDDLs": [
                        ],
                        "unresolvedGroups": [
                        ],
                        "synced": false,
                        "binlogType": "remote",
                        "secondsBehindMaster": "0",
                        "blockDDLOwner": "",
                        "conflictMsg": ""
                    },
                    "validation": {
                        "task": "tidbmigration-sample-task",
                        "source": "tidbmigration-sample-src",
                        "mode": "full",
                        "stage": "Stopped",
                        "validatorBinlog": "(mysql-bin-changelog.058161, 36301019)",
                        "validatorBinlogGtid": "",
                        "result": {
                            "isCanceled": false,
                            "errors": [
                                {
                                    "ErrCode": 43003,
                                    "ErrClass": "validator",
                                    "ErrScope": "internal",
                                    "ErrLevel": "high",
                                    "Message": "failed to get event",
                                    "RawCause": "Header \u0026replication.EventHeader{Timestamp:0x63eecd70, EventType:0x20, ServerID:0x35a66d02, EventSize:0x47, LogPos:0x250bc62, Flags:0x0}, Data \"....\", Err: table id 140: invalid table id, no corresponding table map event",
                                    "Workaround": ""
                                }
                            ],
                            "detail": null
                        },
                        "processedRowsStatus": "insert/update/delete: 405548/181636/182125",
                        "pendingRowsStatus": "insert/update/delete: 3527/1545/1533",
                        "errorRowsStatus": "new/ignored/resolved: 3/0/0"
                    }
                }
            ]
        }
    ]
}```

**Comment by D3Hunter** [comment]
right now there's auto-resume feature for continuous data validation, if it stopped by some resumable error, you can start it again by [`dmctl validation start <task-name>`](https://docs.pingcap.com/tidb/dev/dm-continuous-data-validation#method-2-enable-using-dmctl), it will resume from previous saved location.

**Comment by D3Hunter** [comment]
closed, since 6.2.0 is a DMR version which don't have following fix version
